presto-cli/src/test/java/com/facebook/presto/cli/AbstractCliTest.java

```diff
@@ -0,0 +1,146 @@
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.airlift.json.JsonCodec.jsonCodec;
+import static com.facebook.presto.cli.ClientOptions.OutputFormat.CSV;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static org.testng.Assert.assertEquals;
+
+public abstract class AbstractCliTest
+{
+    protected static final JsonCodec<QueryResults> QUERY_RESULTS_JSON_CODEC = jsonCodec(QueryResults.class);
+
+    protected MockWebServer server;
+
+    @BeforeMethod
+    public void setup()
+            throws IOException
+    {
+        server = new MockWebServer();
+        server.start();
+    }
+
+    @AfterMethod
+    public void teardown()
+            throws IOException
+    {
+        server.close();
+    }
+
+    protected ClientSession createMockClientSession()
+    {
+        return new ClientSession(
+                server.url("/").uri(),
+                "user",
+                "source",
+                Optional.empty(),
+                ImmutableSet.of(),
+                "clientInfo",
+                "catalog",
+                "schema",
+                "America/Los_Angeles",
+                Locale.ENGLISH,
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                null,
+                new Duration(2, MINUTES),
+                ImmutableMap.of());
+    }
+
+    protected QueryResults createMockQueryResults()
+    {
+        return new QueryResults(
+                "20160128_214710_00012_rk68b",
+                server.url("/query.html?20160128_214710_00012_rk68b").uri(),
+                null,
+                null,
+                ImmutableList.of(new Column("_col0", BigintType.BIGINT)),
+                ImmutableList.of(ImmutableList.of(123)),
+                StatementStats.builder().setState("FINISHED").build(),
+                null,
+                ImmutableList.of(),
+                null,
+                null);
+    }
+
+    protected MockResponse createMockResponse()
+    {
+        return new MockResponse()
+                .addHeader(CONTENT_TYPE, "application/json")
+                .setBody(QUERY_RESULTS_JSON_CODEC.toJson(createMockQueryResults()));
+    }
+
+    protected void executeQueries(List<String> queries)
+    {
+        Console console = new Console();
+        QueryRunner queryRunner = createQueryRunnerWithClientSession(createMockClientSession());
+        for (String query : queries) {
+            console.executeCommand(queryRunner, query, CSV, false);
+        }
+    }
+
+    protected static QueryRunner createQueryRunnerWithClientSession(ClientSession clientSession)
```

Comment:
nit: Seems to me the old name `createQueryRunner` was just good enough.

Reply:
Done